### PR TITLE
Command line controls

### PR DESCRIPTION
**Pull request by t3chguy**

### CODE REVIEW

File: src/github.com/t3chguy/riot-static/mxclient/mxclient.go [hunk 1]
```diff
@@ -80,50 +80,58 @@ func (m *Client) forwardpaginateRoom(room *Room, amount int) (int, error) {
...
-		if e != nil {
-			fmt.Printf("File error: %v\n", e)
-			os.Exit(1)
-		}
-
-		json.Unmarshal(file, &config)
+func newClient(homeserverURL, userID, accessToken string) (*Client, error) {
+	cli, err := gomatrix.NewClient(homeserverURL, userID, accessToken)
+	cli.Client = &http.Client{
+		Timeout: 5 * time.Second,
```
Comment (Kegsay) [inline src/github.com/t3chguy/riot-static/mxclient/mxclient.go]: This is probably too low for Synapse. I'd up this to 30s.

**Comment by Kegsay** [comment]
LGTM

re:
> whether or not my choices for command line args etc are appropriate

Config options are nearly always subjective. So long as they are coherent and consistent then there shouldn't be any problems. This PR is a little bit yucky because you have a conditional flag that is only "active" when you specify another flag. Sure, you could split this out into a separate binary, but given you only have 3 options, having it like this is perfectly fine. If there were a lot of options I would be suggesting otherwise though.

**Comment by t3chguy** [comment]
@Kegsay how about making `--register-guest` take a homeserver URL?

**Comment by Kegsay** [comment]
Sure.

**Comment by t3chguy** [comment]
I'll keep it like it is for now, the alternative is equally nasty imho

**Pull request merged**